arelle/WebCache.py

```diff
@@ -335,6 +344,16 @@ def getfilename(self, url, base=None, reload=False, checkModifiedTime=False, nor
                 return None
             # form cache file name (substituting _ for any illegal file characters)
             filepath = self.urlToCacheFilepath(url)
+            if self.httpsRedirect:
+                if not os.path.exists(filepath):
+                    # if enabled, check for missing files in their inverse http/https cache directory
+                    redirect = None
+                    if url.startswith('http://'):
```

Comment:
I think that `urlparse()` call does quite a bit more than we need it to. Some logic up above already guarantees that the scheme is either http or https so I think it's a bit more lightweight to skip the full URL parsing and just do this quick swap.

Reply:
I would have guessed that `os.path.exists` call would dominate here.  It's a bit unfortunate that it can't somehow be combined with the one below so that we only need one filesystem access rather than two in the (common?) case where there is no mismatch.